### ISSUE TITLE
On Bash 5.x, Autopkg puts itself in Background

### ISSUE THREAD

**Issue by workflowsguy** [autopkg/autopkg#454]
Since upgrading Bash to 5.0 on my Mac, autopkg has changed it's behavior. After starting it with a command line like `autopkg run -vvvv -k DISABLE_CODE_SIGNATURE_VERIFICATION=1 Fritzing.munki`, the program puts itself in the background.
I have to enter `fg` repeatedly (5 times at least) to get it back to the foreground.

I could not find anything about this in the documentation, so I do not know if this is "by design" or an error; I consider it problematic.

autopkg 1.0.4
macOS 10.13.6
Bash 5.0.2(1)-release

**Comment by gregneagle** [comment]
I've never seen this behavior and have seen no other reports of it with the bash that ships with macOS. Of course, macOS does not ship with Bash 5.0, and the autopkg developers probably should not be held responsible for testing with shells installed from random internet sources...

You should consider posting on in the autopkg-discuss group: http://groups.google.com/group/autopkg-discuss

**Comment by gregneagle** [comment]
I just downloaded the bash-5.0 source from http://ftp.gnu.org/gnu/bash/ and compiled it. I then ran `autopkg` under bash-5.0 and did not see the behavior you describe:

```
bash-5.0$ autopkg run -vvv munkitools3.munki
Processing munkitools3.munki...
{'AUTOPKG_VERSION': u'1.0.4',
 u'GIT_PATH': u'/usr/local/bin/git',
 u'INCLUDE_PRERELEASES': u'False',
 u'MUNKILIB_DIR': u'/Users/Shared/munki-git/munki/code/client',
 u'MUNKITOOLS_ADMIN_DESCRIPTION': u'Command-line Managed Software Center admin tools.',
 u'MUNKITOOLS_ADMIN_DISPLAYNAME': u'Managed Software Center admin tools',
 u'MUNKITOOLS_ADMIN_NAME': u'munkitools_admin',
 u'MUNKITOOLS_APP_DESCRIPTION': u'Managed Software Center application.',
 u'MUNKITOOLS_APP_DISPLAYNAME': u'Managed Software Center',
 u'MUNKITOOLS_APP_NAME': u'munkitools',
<snip>
Receipt written to /Users/gneagle/Library/AutoPkg/Cache/local.munki.munkitools3/receipts/munkitools3-receipt-20190317-163453.plist

The following new items were imported into Munki:
    Name        Version     Catalogs  Pkginfo Path                            Pkg Repo Path                             
    ----        -------     --------  ------------                            -------------                             
    munkitools  5.1.0.3733  testing   munkitools/munkitools-5.1.0.3733.plist  munkitools/munkitools_app-5.1.0.3733.pkg  

The following new items were downloaded:
    Download Path                                                                                     
    -------------                                                                                     
    /Users/gneagle/Library/AutoPkg/Cache/local.munki.munkitools3/downloads/munkitools-3.6.0.3733.pkg  
```

**Comment by gregneagle** [comment]
Any follow up here?

**Comment by workflowsguy** [comment]
I changed the Bash shell back to the 3.x version that comes with macOS (5.x was installed through Homebrew, btw.).
However, I am now also seeing the behaviour with this old Bash version. So, Bash version does not seem to be the cause.

I also searched the internet for reports that might report a similar issue, but so far only got results that describe how to put programs in the background manually (Shell) or through programming (Python).

I will next try to find another program written in Python 2 and see if this also shows the unexpected behaviour.

**Comment by gregneagle** [comment]
I'm going to close this as unreproducible outside of your machine. Feel free to reopen if you discover more information or can describe a reproducible case.

**Comment by workflowsguy** [comment]
By luck, I discovered what caused this problem (it had nothing to do with the Bash shell version):

Some process has changed the owner of several packages in `/usr/local/lib/python3.7/site-packages` to `root`.

After I changed this back by running `sudo chown -R $(whoami) /usr/local/lib/python3.7/site-packages`, all reverted back to normal. This also solved the `hdiutil` error mentioned in #455.

**Comment by gregneagle** [comment]
I'd think the owners of everything /usr/local/lib/* should be root.

Additionally, autopkg is not supported with Python 3.x.

You have odd things going on on your machine!
